clpy/core/core.pyx

```diff
@@ -3074,23 +3074,26 @@ cpdef _scatter_op_single(ndarray a, ndarray indices, v,
         (1,) * len(lshape) + indices_shape + (1,) * len(rshape))
     indices = broadcast_to(indices, v_shape)
 
-    raise NotImplementedError("clpy does not support this")
-#    if op == 'update':
-#        _scatter_update_kernel(
-#            v, indices, cdim, rdim, adim, a.reduced_view())
-#    elif op == 'add':
-#        # There is constraints on types because atomicAdd() in CUDA 7.5
-#        # only supports int32, uint32, uint64, and float32.
-#        if not issubclass(v.dtype.type,
-#                          (numpy.int32, numpy.float32,
-#                           numpy.uint32, numpy.uint64, numpy.ulonglong)):
-#            raise TypeError(
-#                'scatter_add only supports int32, float32, uint32, uint64 as '
-#                'data type')
-#        _scatter_add_kernel(
-#            v, indices, cdim, rdim, adim, a.reduced_view())
-#    else:
-#        raise ValueError('provided op is not supported')
+    if op == 'update':
+        # _scatter_update_kernel(
+        #     v, indices, cdim, rdim, adim, a.reduced_view())
+        raise NotImplementedError("clpy does not support this")
+    elif op == 'add':
+        # There is constraints on types because atomicAdd() in CUDA 7.5
+        # only supports int32, uint32, uint64, and float32.
+        # if not issubclass(v.dtype.type,
+        #                   (numpy.int32, numpy.float32,
+        #                    numpy.uint32, numpy.uint64, numpy.ulonglong)):
+        #     raise TypeError(
+        #         'scatter_add only supports int32, float32, uint32, uint64 as'
+        #         ' data type')
+        # TODO(vorj): fix this after resolve #30
+        if not issubclass(v.dtype.type, numpy.float32):
+            raise NotImplementedError('clpy does not support this')
```

Comment:
How about his?

Reply:
CuPy has [ `scatter_add` test only for float32 ](https://github.com/fixstars/clpy/blob/e3db666bf8399db34dd66069af9e8c2fd225afba/tests/clpy_tests/ext_tests/test_scatter.py#L16) and current ClPy also does.
Some tests like `test_ndarray_scatter.py` has test of `scatter_add` for other types, so it might be said that it is test of this pass to fail `test_ndarray_scatter.py` .

Reply:
OK. This `NotImplementedError` could be removed by #75 .